core/Plugin/Controller.php

```diff
@@ -476,25 +476,18 @@ protected function getLastUnitGraphAcrossPlugins($currentModuleName, $currentCon
      */
     protected function getGraphParamsModified($paramsToSet = array())
     {
-        if (!isset($paramsToSet['period'])) {
-            $period = Common::getRequestVar('period');
-        } else {
-            $period = $paramsToSet['period'];
+        try {
+            $period = isset($paramsToSet['period']) ?? Piwik::getPeriod();
+        } catch (\Exception $e) {
+            $period = null;
         }
+
         if ($period === 'range') {
             return $paramsToSet;
         }
-        if (!isset($paramsToSet['range'])) {
-            $range = 'last30';
-        } else {
-            $range = $paramsToSet['range'];
-        }
 
-        if (!isset($paramsToSet['date'])) {
-            $endDate = $this->strDate;
-        } else {
-            $endDate = $paramsToSet['date'];
-        }
+        $range = isset($paramsToSet['range']) ?? 'last30';
```

Comment:
This won't have the result it should have. Same below.
It's either `isset($x) ? $x : 'default'` or `$x ?? 'default'`